eZ/Publish/API/Repository/Tests/UserServiceTest.php

```diff
@@ -1576,6 +1578,80 @@ public function testUpdateUser()
     /**
      * Test for the updateUser() method.
      *
+     * @return \eZ\Publish\API\Repository\Values\User\User
+     *
+     * @see \eZ\Publish\API\Repository\UserService::updateUser()
+     * @depends eZ\Publish\API\Repository\Tests\UserServiceTest::testCreateUser
+     * @depends eZ\Publish\API\Repository\Tests\UserServiceTest::testNewUserUpdateStruct
+     * @depends eZ\Publish\API\Repository\Tests\ContentServiceTest::testUpdateContent
+     * @depends eZ\Publish\API\Repository\Tests\ContentServiceTest::testUpdateContentMetadata
+     */
+    public function testUpdateUserNoPassword()
+    {
+        $repository = $this->getRepository();
+        $signalSlotUserService = $repository->getUserService();
+
+        $signalSlotUserServiceReflection = new ReflectionClass($signalSlotUserService);
+        $userServiceProperty = $signalSlotUserServiceReflection->getProperty('service');
+        $userServiceProperty->setAccessible(true);
+        $userService = $userServiceProperty->getValue($signalSlotUserService);
+
+        $userServiceReflection = new ReflectionClass($userService);
+        $settingsProperty = $userServiceReflection->getProperty('settings');
+        $settingsProperty->setAccessible(true);
+        $settingsProperty->setValue(
+            $userService,
+            [
+                'hashType' => User::PASSWORD_HASH_MD5_USER,
```

Comment:
Can we move these constants to API? They kind of need to stay constant so could make sense there and would avoid adding more hard coding of Core logic in integration tests _(they are not supposed to known about Core internals)_. /cc @alongosz